x/leverage/gov_handler.go

```diff
@@ -22,12 +22,10 @@ func NewUpdateRegistryProposalHandler(k keeper.Keeper) govtypes.Handler {
 }
 
 func handleUpdateRegistryProposalHandler(ctx sdk.Context, k keeper.Keeper, p *types.UpdateRegistryProposal) error {
-	if err := k.DeleteRegisteredTokens(ctx); err != nil {
-		return err
-	}
-
 	for _, token := range p.Registry {
-		k.SetRegisteredToken(ctx, token)
+		if err := k.SetRegisteredToken(ctx, token); err != nil {
+			return err
+		}
 	}
```

Comment:
Note the functional difference in update token registry - now any existing tokens will remain or be updated, instead of being deleted. It is no longer possible to delete registered tokens entirely.